src/simple_minreq.rs

```diff
@@ -0,0 +1,271 @@
+//! This module implements the [`crate::client::Transport`] trait using [minreq] as the underlying
+//! HTTP transport.
+//!
+//! [minreq]: <https://github.com/neonmoe/minreq>
+//!
+
+#[cfg(not(fuzzing))]
+use std::time::Duration;
+use std::{error, fmt};
+
+use crate::client::Transport;
+use crate::{Request, Response};
+
+/// Global mutex used by the fuzzing harness to inject data into the read
+/// end of the TCP stream.
+#[cfg(fuzzing)]
+pub static FUZZ_TCP_SOCK: Mutex<Option<io::Cursor<Vec<u8>>>> = Mutex::new(None);
+
+#[cfg(fuzzing)]
+#[derive(Clone, Debug)]
+struct TcpStream;
+
+#[cfg(fuzzing)]
+mod impls {
+    use super::*;
+    impl Read for TcpStream {
+        fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
+            match *FUZZ_TCP_SOCK.lock().unwrap() {
+                Some(ref mut cursor) => io::Read::read(cursor, buf),
+                None => Ok(0),
+            }
+        }
+    }
+    impl Write for TcpStream {
+        fn write(&mut self, buf: &[u8]) -> io::Result<usize> {
+            io::sink().write(buf)
+        }
+        fn flush(&mut self) -> io::Result<()> {
+            Ok(())
+        }
+    }
+
+    impl TcpStream {
+        pub fn connect_timeout(_: &SocketAddr, _: Duration) -> io::Result<Self> {
+            Ok(TcpStream)
+        }
+        pub fn set_read_timeout(&self, _: Option<Duration>) -> io::Result<()> {
+            Ok(())
+        }
+        pub fn set_write_timeout(&self, _: Option<Duration>) -> io::Result<()> {
+            Ok(())
+        }
+    }
+}
+
+/// The default TCP port to use for connections.
+/// Set to 8332, the default RPC port for bitcoind.
+pub const DEFAULT_PORT: u16 = 8332;
+
+/// Simple HTTP transport that implements the necessary subset of HTTP for
+/// running a bitcoind RPC client.
+#[derive(Clone, Debug)]
+pub struct SimpleHttpTransport {
+    url: String,
+    timeout: Duration,
+    /// The value of the `Authorization` HTTP header, i.e., a base64 encoding of 'user:password'.
+    basic_auth: Option<String>,
+}
+
+impl Default for SimpleHttpTransport {
+    fn default() -> Self {
+        SimpleHttpTransport {
+            url: format!("localhost:{}", DEFAULT_PORT),
```

Comment:
In d20ba2eafcaa3cd5245f5a548cbcbb6d14c13432:

minreq seems to require a `http://` prefix on this or else you get the error "was redirected to an absolute url with an invalid protocol".

Reply:
I"ll add integration testing against a bitcoind instance like we do in `rust-bitcoincorerpc` eh, to catch this sort of mistake.

Reply:
That would be amazing. TBH I usually test nontrivial PRs  to this crate by syncing my hot wallet with it, which is probably not the safest testing strategy..

Reply:
Added `http://` to the string, FTR I have not tested this code path.